ClosedXML/Excel/CalcEngine/Functions/Lookup.cs

```diff
@@ -87,6 +88,84 @@ private static object Hyperlink(List<Expression> p)
             return new XLHyperlink(address, toolTip);
         }
 
+        private static object Match(List<Expression> p)
+        {
+            var lookup_value = p[0];
+            int match_type = 1;
+            if (p.Count > 2)
+                match_type = Math.Sign((int)p[2]);
+
+            if (!(p[1] is XObjectExpression objectExpression))
+                throw new NoValueAvailableException("lookup_array has to be a range");
+
+            if (!(objectExpression.Value is CellRangeReference cellRangeReference))
+                throw new NoValueAvailableException("lookup_array has to be a range");
+
+            var range = cellRangeReference.Range;
+
+            if (range.ColumnCount() != 1 && range.RowCount() != 1)
+                throw new CellValueException("Range has to be 1-dimensional");
+
+            Predicate<int> lookupPredicate = null;
+            switch (match_type)
+            {
+                case 0:
+                    lookupPredicate = i => i == 0;
+                    break;
+
+                case 1:
+                    lookupPredicate = i => i <= 0;
+                    break;
+
+                case -1:
+                    lookupPredicate = i => i >= 0;
+                    break;
+
+                default:
+                    throw new NoValueAvailableException("Invalid match_type");
+            }
+
+            IXLCell foundCell = null;
+
+            if (match_type == 0)
+                foundCell = range
+                    .CellsUsed(XLCellsUsedOptions.Contents, c => !c.IsEmpty() && lookupPredicate.Invoke(new Expression(c.Value).CompareTo(lookup_value)))
+                    .FirstOrDefault();
+            else
+            {
+                object previousValue = null;
+                foundCell = range
+                    .CellsUsed(XLCellsUsedOptions.Contents)
+                    .TakeWhile(c =>
+                    {
+                        if (c.IsEmpty())
+                            return false;
+
+                        var currentCellExpression = new Expression(c.Value);
+
+                        if (previousValue != null)
+                        {
+                            // When match_type != 0, we have to assume that the order of the items being search is ascending or descending
+                            var previousValueExpression = new Expression(previousValue);
+                            if (!lookupPredicate.Invoke(previousValueExpression.CompareTo(currentCellExpression)))
+                                return false;
+                        }
+
+                        previousValue = c.Value;
+
+                        return lookupPredicate.Invoke(currentCellExpression.CompareTo(lookup_value));
```

Comment:
I think, instead of defining `lookup_predicate` we can preserve a simple int (-1, 0 or 1) and compare the result with it (`return currentCellExpression.CompareTo(lookup_value) == expectedComparisonResult`).

Reply:
But when `match_type != 0`, then `lookup_predicate` uses '<=', not '<', so a single `int` won't be enough.

Reply:
Ah, I missed that.

Reply:
Also, the result can be other values too, e.g. `StringComparer.OrdinalIgnoreCase.Compare("asdf", "qwer") == -16`